reindexer/reindex_worker/src/test/scala/uk/ac/wellcome/platform/reindex/reindex_worker/dynamo/BatchItemGetterTest.scala

```diff
@@ -0,0 +1,63 @@
+package uk.ac.wellcome.platform.reindex.reindex_worker.dynamo
+
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.platform.reindex.reindex_worker.fixtures.ReindexDynamoFixtures
+
+class BatchItemGetterTest
+    extends FunSpec
+    with Matchers
+    with ScalaFutures
+    with ReindexDynamoFixtures {
+
+  it("finds a single specified record") {
+    withLocalDynamoDbTable { table =>
+      val batchItemGetter = createBatchItemGetter
+
+      val records = createRecords(table, count = 5)
+      val specifiedRecord = records.head
+
+      val futureResult =
+        batchItemGetter.get(List(specifiedRecord.id))(table.name)
+
+      whenReady(convertScalaFuture(futureResult)) { result =>
+        convertToAnyShouldWrapper(
+          result.map { fromJson[NamedRecord](_).get }.head shouldBe specifiedRecord)
+      }
+    }
+  }
+
+  it("finds a list of specified records") {
+    withLocalDynamoDbTable { table =>
+      val batchItemGetter = createBatchItemGetter
+
+      val records = createRecords(table, count = 5)
+      val specifiedRecords = records.slice(1, 3)
+
+      val futureResult =
+        batchItemGetter.get(specifiedRecords.map { _.id }.toList)(table.name)
+
+      whenReady(convertScalaFuture(futureResult)) { result =>
+        convertToAnyShouldWrapper(
+          result
+            .map { fromJson[NamedRecord](_).get } shouldEqual specifiedRecords)
+      }
+    }
+  }
+
+  it("handles being asked for a non-existent record") {
+    withLocalDynamoDbTable { table =>
+      val batchItemGetter = createBatchItemGetter
+
+      createRecords(table, count = 5)
+
+      val futureResult =
+        batchItemGetter.get(List("bananas"))(table.name)
+
+      whenReady(convertScalaFuture(futureResult)) { result =>
+        convertToAnyShouldWrapper(result shouldBe empty)
+      }
+    }
+  }
```

Comment:
Some other test cases you could cover:

* What if I ask for a list of IDs, and some of them exist and some don't? (You could argue that returning the ones that exist and ignoring the rest is acceptable, or that errors should never pass silently so you should error out. I don't think it really matters, but a test to check which convention we're using would be helpful.)
* What if the table doesn't exist?

Reply:
There is definitely some work here around defining what is a Success and Failure of the service, one issue with this is we don't surface it at all - so it's pretty useless building those What ifs in.

Reply:
Have left out the table existence test for both the above reason and because it's not included in existing/equivalent tests. Definitely worth revisiting if we reindex more frequently etc etc